### PR TITLE
Added possibility to pass tests script as string to address cases whe…

### PR DESCRIPTION
**Pull request by Chekusty**
Added possibility to pass tests script as string to address cases when powershell scripts and tests for them are stored not on FileSystem but in some storage like DataBase and executed in Temp folder.

Example is Sitecore CMS, where we use SPE (Sitecore Powershell Extensions) module to create and run PS scripts. We store scripts and tests as Items in content tree and SPE executes them when needed in Temp folder on the server.

Will try to add Pester tests later this week.

### CODE REVIEW

File: Pester.psm1 [hunk 1]
```diff
@@ -829,45 +835,59 @@ New-PesterOption
                 param (
                     [Parameter(Position = 0)]
                     [string] $Path,
-
+                    [string] $Script, #in order to allow passing test script as a string
```
Comment (it-praktyk) [inline Pester.psm1]: An added parameter is not explained in the comment based help.

File: Pester.psm1 [hunk 2]
```diff
@@ -829,45 +835,59 @@ New-PesterOption
...
                     [object[]] $Arguments = @(),
                     [System.Collections.IDictionary] $Parameters = @{}
                 )
 
-                & $Path @Parameters @Arguments
+                if(![string]::IsNullOrEmpty($Path))
```
Comment (it-praktyk) [inline Pester.psm1]: Please use `-not` instead `!`

File: Pester.psm1 (hunk 1 above)
Comment (Chekusty) [reply Pester.psm1]: I removed it from here. Mentioned usage of this parameter in example (line 661). Do you think it understandable enough?

File: Pester.psm1 (hunk 2 above)
Comment (Chekusty) [reply Pester.psm1]: Done! Thanks!

File: Pester.psm1 (hunk 1 above)
Comment (Chekusty) [reply Pester.psm1]: @it-praktyk Could you please review my PR again? I`ve added test and description for parameter. Changes are backward compatible.

**Comment by nohwnd** [comment]
@Chekusty It fails on the build server on PowerShell 2 with this error: 
```
[21:05:13][Step 1/1] C:\TeamCity\BuildAgent\temp\buildTmp\powershell7172863267207089627.ps1 : Proper
[21:05:13][Step 1/1] ty 'ScriptStackTrace' cannot be found on this object. Make sure that it exists.
[21:05:13][Step 1/1]     + CategoryInfo          : InvalidOperation: (.:OperatorToken) [Write-Error 
[21:05:13][Step 1/1]    ], RuntimeException
[21:05:13][Step 1/1]     + FullyQualifiedErrorId : PropertyNotFoundStrict,powershell717286326720708 
[21:05:13][Step 1/1]    9627.ps1
[21:05:13][Step 1/1]  
[21:05:13][Step 1/1] Process exited with code 1
[21:05:13][Step 1/1] Process exited with code 1
[21:05:14][Step 1/1] Step Run Tests (PowerShell) failed
[21:05:14]NUnit report watcher
[21:05:14][NUnit report watcher] No reports found for paths:
[21:05:14][NUnit report watcher] +:Test.*.xml
```

**Comment by Chekusty** [comment]
@nohwnd I think I need a little bit of your help, because I cannot find error locally, and error message is not descriptive.. Do you have any idea what can be wrong?

**Comment by it-praktyk** [comment]
@Chekusty, it's internal error of the TeamCity instance.

**Comment by Chekusty** [comment]
@it-praktyk Ok. Can I expect that someone will fix this internal error?

**Comment by it-praktyk** [comment]
I hope. I've [asked](https://github.com/pester/Pester/pull/1013#issuecomment-377377355) yesterday @dlwyatt for that.

**Comment by Chekusty** [comment]
@it-praktyk Great! Thank you, guys!

**Comment by nohwnd** [comment]
@it-praktyk  Great. That is the line I thought will be the case. Weird that it does not print the whole error message in the TC output, I will try to enhance that first, and then fix this error.

**Comment by nohwnd** [comment]
I could not push into your master, so I am merging this as is, and will merge a fix for the problems there were on PowerShell 2 shortly after. Thanks for the change. 

Next time please make a branch for the change instead of making a pull-request from your master, if you don't know how to do it please ask 🙂

**Comment by Chekusty** [comment]
OMG, thank you very much! I lost my hope this summer :-) Now we will finally be able to switch to Pester, not ChekustyCustomPester :-) Thank you!

**Comment by nohwnd** [comment]
@Chekusty Sorry it took so long, I will release it in few days. 😊

**Pull request merged**